tests/scalers/neo4j/neo4j_test.go

```diff
@@ -0,0 +1,296 @@
+
+	. "github.com/kedacore/keda/v2/tests/helper"
+)
+
+// Load environment variables from .env file
+var _ = godotenv.Load("../../.env")
+
+const (
+	testName = "neo4j-test"
+)
+
+var (
+	scalerName       = "test-release"
+	testNamespace    = fmt.Sprintf("%s-ns", testName)
+	protocol         = "neo4j"
+	neo4jPassword    = "mypassword"
+	deploymentName   = fmt.Sprintf("%s-deployment", testName)
+	secretName       = fmt.Sprintf("%s-secret", testName)
+	triggerAuthName  = fmt.Sprintf("%s-ta", testName)
+	scaledObjectName = fmt.Sprintf("%s-sobj", testName)
+	neo4jNamespace   = "neo4j-ns"
+	neo4jUser        = "neo4j"
+	neo4jHelmRepo    = "https://helm.neo4j.com/neo4j"
+	minReplicaCount  = 0
+	maxReplicaCount  = 2
+)
+
+type templateData struct {
+	TestNamespace    string
+	PasswordArg      string
+	UsernameArg      string
+	ScalerName       string
+	Protocol         string
+	DeploymentName   string
+	HostName         string
+	Port             string
+	Username         string
+	Password         string
+	Neo4jNamespace   string
+	SecretName       string
+	TriggerAuthName  string
+	ScaledObjectName string
+	MinReplicaCount  int
+	MaxReplicaCount  int
+}
+
+const (
+	deploymentTemplate = `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{.DeploymentName}}
+  namespace: {{.TestNamespace}}
+  labels:
+    app: {{.DeploymentName}}
+spec:
+  replicas: 0
+  selector:
+    matchLabels:
+      app: {{.DeploymentName}}
+  template:
+    metadata:
+      labels:
+        app: {{.DeploymentName}}
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:alpine
+`
+
+	secretTemplate = `
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{.SecretName}}
+  namespace: {{.TestNamespace}}
+data:
+  username: {{.Username}}
+  password: {{.Password}}
+`
+	triggerAuthTemplate = `apiVersion: keda.sh/v1alpha1
+kind: TriggerAuthentication
+metadata:
+  name: {{.TriggerAuthName}}
+  namespace: {{.TestNamespace}}
+spec:
+  secretTargetRef:
+  - parameter: password
+    name: {{.SecretName}}
+    key: password
+  - parameter: username
+    name: {{.SecretName}}
+    key: username
+`
+	scaledObjectTemplate = `
+apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: {{.ScaledObjectName}}
+  namespace: {{.TestNamespace}}
+spec:
+  scaleTargetRef:
+    name: {{.DeploymentName}}
+  minReplicaCount: {{.MinReplicaCount}}
+  maxReplicaCount: {{.MaxReplicaCount}}
+  triggers:
+  - type: neo4j
+    metadata:
+      host: {{.HostName}}
+      protocol: {{.Protocol}}
+      port: "7687"
+      queryValue: "9"
+      query: 'MATCH (n:Person)<-[r:FOLLOWS]-() WHERE n.popularfor IS NOT NULL RETURN n,COUNT(r) order by COUNT(r) desc LIMIT 1'
+      activationQueryValue: "9"
+    authenticationRef:
+      name: {{.TriggerAuthName}}
+`
+)
+
+func CreateKubernetesResourcesNeo4j(t *testing.T, kc *kubernetes.Clientset, nsName string, data interface{}, templates []Template) {
+	KubectlApplyMultipleWithTemplate(t, data, templates)
+}
```

Comment:
What is this function for? I mean, it's just a wrapper

Reply:
Yes, it is just a wrapper